lib/graph_gateway.go

```diff
@@ -315,14 +316,46 @@ func (api *GraphGateway) loadRequestIntoSharedBlockstoreAndBlocksGateway(ctx con
 			metrics.contextAlreadyCancelledMetric.Inc()
 		}
 
-		err := api.fetcher.Fetch(ctx, path, func(resource string, reader io.Reader) error {
+		cctx, cncl := context.WithCancel(ctx)
+		defer cncl()
+		err := api.fetcher.Fetch(cctx, path, func(resource string, reader io.Reader) error {
 			cr, err := car.NewCarReader(reader)
 			if err != nil {
 				return err
 			}
+
+			blkCh := make(chan blocks.Block, 1)
+			go func() {
+				defer close(blkCh)
+				for {
+					blk, rdErr := cr.Next()
+					if rdErr != nil {
+						err = rdErr
+					}
+					blkCh <- blk
+				}
+			}()
+
+			t := time.NewTimer(30 * time.Second)
```

Comment:
:thought_balloon:    timeout for  `?format=raw` fetch across all retries  is 1m (`GetBlockTimeout`).
Switching to `?format=car` increased fetch timeout to 30m, so we are waiting longer than 1m for the first block.

Changing it to 30s means we will be failing to fetch the first block more often than the old GRAPH_BACKEND=false.
We could start with the same timeout as for `?format=raw` for now,

```suggestion
			t := time.NewTimer(GetBlockTimeout)
```

@willscott thoughts?

Reply:
makes sense 👍 changed here and below on line 354